runtime/caml/platform.h

```diff
@@ -120,6 +92,231 @@ void caml_plat_broadcast(caml_plat_cond*);
 void caml_plat_signal(caml_plat_cond*);
 void caml_plat_cond_free(caml_plat_cond*);
 
+/* Futexes */
+
+/* An integer that can be waited on and woken, used to build other
+   synchronisation primitives; either uses OS facilities directly, or
+   a condition-variable fallback */
+typedef struct caml_plat_futex caml_plat_futex;
+
+typedef uint32_t caml_plat_futex_value;
+typedef _Atomic caml_plat_futex_value caml_plat_futex_word;
+
+/* Block while `futex` has the value `undesired`, until woken by `wake_all` */
+void caml_plat_futex_wait(caml_plat_futex* futex,
+                          caml_plat_futex_value undesired);
+/* Wake all threads `wait`ing on `futex` */
+void caml_plat_futex_wake_all(caml_plat_futex* futex);
+
+/* Define CAML_PLAT_FUTEX_FALLBACK to use the condition-variable
+   fallback, even if a futex implementation is available */
+#ifndef CAML_PLAT_FUTEX_FALLBACK
+#  if defined(_WIN32)                                   \
+  || (defined(__linux__) && defined(HAS_LINUX_FUTEX_H)) \
+  || defined(__FreeBSD__) || defined(__OpenBSD__)
+/*  These exist, but are untested
+     defined(__NetBSD__) || defined(__DragonFly__) */
+#  else
+#    define CAML_PLAT_FUTEX_FALLBACK
+#  endif
+#endif
+
+#if !defined(CAML_PLAT_FUTEX_FALLBACK)
+struct caml_plat_futex {
+  caml_plat_futex_word value;
+};
+#  define CAML_PLAT_FUTEX_INITIALIZER(value) { (value) }
+
+Caml_inline void caml_plat_futex_init(caml_plat_futex* ftx,
```

Comment:
They are both trivial for this case, and I feel like it would be a waste to have them actually be emitted as functions, though I'm not vehemently opposed- it might make the code slightly clearer.